src/Common/Services/Mail/PostmarkMailProvider.cs

```diff
@@ -2,33 +2,24 @@
 
 namespace Passwordless.Common.Services.Mail;
 
-public class PostmarkMailProvider : IMailProvider
+public class PostmarkMailProvider(PostmarkMailClientFactory factory) : IMailProvider
 {
```

Comment:
suggestion: I'm wondering if we could remove the factory stuff and but the logic in the constructor here? Seems like we could get away with something simple and more direct:

// setup the default client
_defaultClient = defaultClient

foreach stream in Config.messageStreams
  _streamClients.add(stream.name, streamClient


and then in Send you do a streamclient.FirstOrDefault() ?? _defaultClient;

Reply:
```
public class PostmarkMailProvider(PostmarkMailProviderConfiguration configuration) : IMailProvider
{
    private readonly Dictionary<string, ConfiguredPostmarkClient> _clients = configuration.MessageStreams?
        .ToDictionary(x => x.Name, GetConfiguredClient) ?? new();

    private readonly ConfiguredPostmarkClient _defaultClient = GetConfiguredClient(configuration.DefaultConfiguration);
    
    public async Task SendAsync(MailMessage message)
    {
        var configuredClient = _clients.GetValueOrDefault(message.MessageType) ?? _defaultClient;
        
        PostmarkMessage pm = new PostmarkMessage
        {
            To = string.Join(',', message.To),
            From = configuredClient.From?.ToString() ?? message.From,
            Subject = message.Subject,
            TextBody = message.TextBody,
            HtmlBody = message.HtmlBody,
            Tag = message.Tag,
            Bcc = message.Bcc.Count != 0 ? string.Join(',', message.Bcc) : null,
            MessageStream = configuredClient == _defaultClient ? null : message.MessageType
        };

        IEnumerable<PostmarkResponse>? res = await configuredClient.Client.SendMessagesAsync(pm);
    }
    
    private static ConfiguredPostmarkClient GetConfiguredClient(PostmarkClientConfiguration config) =>
        new()
        {
            Client = new PostmarkClient(config.ApiKey),
            From = new MailAddress(config.From),
            Name = config.Name
        };
}
```

Here's what PostmarkMailProvider becomes.

Pros:
- Reduces everything to 1 file.

Cons:
- all clients are instantiated when the provider is created. (not a big deal now, but could be if we decide to create more of them)
- allows for a misconfiguration to go unnoticed (if a message type gets set to something that isn't configured, it will use the default and we would only notice by checking the from after the email has been sent out.

Reply:
We could go a bit further and move the mapping from the config file straight to the strongly typed version containing the postmark client in the DI setup code as well and just inject that object.

That way we would already have the dictionary set up and wouldn't need to do it here.

Reply:
I really like the first approach, keeping it in the PostmarkMailProvider.

> Allows for a misconfiguration to go unnoticed (if a message type gets set to something that isn't configured, it will use the default and we would only notice by checking the from after the email has been sent out.

Could we add warning logging for this?

Reply:
Went with option 1 and added a Warning log to the provider in the event that the message type is provided for a message stream we don't currently have a config for.